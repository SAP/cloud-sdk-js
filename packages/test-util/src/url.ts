import { ErrorWithCause } from '@sap-cloud-sdk/util';
import { executeHttpRequest } from '@sap-cloud-sdk/core';

/**
 * Checks whether a URL is existing via a get request. Throws an exception if not.
 * @param url - URL to be checked
 */
export async function checkUrlExists(url: string): Promise<void> {
  try {
    const response = await executeHttpRequest({ url }, { method: 'get' });
    if (response.status !== 200) {
      throw new Error(
        `Reques to ${url} failed with status: ${response.status}`
      );
    }
  } catch (err) {
    throw new ErrorWithCause(`Request to ${url} failed.`, err);
  }
}
