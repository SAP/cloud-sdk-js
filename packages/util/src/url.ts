import axios from 'axios';

/**
 * Checks whether a URL is existing via a head request.
 * @param url - URL to be checked.
 * @returns Promise - resolves if the URL exists.
 */
export async function checkUrlExists(url: string): Promise<number> {
  return axios
    .request({ url, method: 'HEAD' })
    .then(response => response.status);
}
