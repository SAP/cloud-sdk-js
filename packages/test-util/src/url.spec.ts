import { checkUrlExists } from './url';

describe('url', () => {
  it('[E2E] fails for non existing link', async () => {
    await expect(
      checkUrlExists(
        'https://sap.github.io/cloud-sdk/docs/js/features/non-existing-features'
      )
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
