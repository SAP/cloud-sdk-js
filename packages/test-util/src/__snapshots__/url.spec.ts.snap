// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`url [E2E] fails for non existing link 1`] = `"Request to https://sap.github.io/cloud-sdk/docs/js/features/non-existing-features failed."`;
